src/essreflectometry/amor/orso.py

```diff
@@ -1,60 +1,79 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
-import platform
-from datetime import datetime
-
-from orsopy import fileio
-
-from .. import __version__
-
-
-def make_orso(
-    owner: fileio.base.Person,
-    sample: fileio.data_source.Sample,
-    creator: fileio.base.Person,
-    reduction_script: str,
-) -> fileio.orso.Orso:
-    """
-    Generate the base Orso object for the Amor instrument.
-    Populate the Orso object for metadata storage.
-
-    Parameters
-    ----------
-    owner:
-        The owner of the data set, i.e. the main proposer of the measurement.
-    sample:
-        A description of the sample.
-    creator:
-        The creator of the reduced data, the person responsible for the
-        reduction process.
-    reduction_script:
-        The script or notebook used for reduction.
-
-    Returns
-    -------
-    :
-        Orso object with the default parameters for the Amor instrument.
-    """
-    orso = fileio.orso.Orso.empty()
-    orso.data_source.experiment.probe = 'neutrons'
-    orso.data_source.experiment.facility = 'Paul Scherrer Institut'
-    orso.data_source.measurement.scheme = 'angle- and energy-dispersive'
-    orso.reduction.software = fileio.reduction.Software(
-        'scipp-ess', __version__, platform.platform()
+"""ORSO utilities for Amor."""
+import os
+from typing import Optional
+
+from dateutil.parser import parse as parse_datetime
+from orsopy.fileio import base as orso_base
+from orsopy.fileio import data_source
+
+from ..orso import (
+    OrsoExperiment,
+    OrsoInstrument,
+    OrsoMeasurement,
+    OrsoOwner,
+    OrsoSample,
+)
+from ..types import Filename, RawData, Reference, Run, Sample
+
+
+def parse_orso_experiment(raw_data: RawData[Run]) -> OrsoExperiment[Run]:
```

Comment:
If you are concerned about keeping alive `RawData` longer than necessary, consider extracting meta data first, or loading neutron data and meta data independently.

Reply:
> consider extracting meta data first, 

How would that work?

> loading neutron data and meta data independently.

I considered it. But that would either mean loading the whole file twice which is slow. Or writing two more complicated loaders.

Reply:
I think it is quite simple to write such a loader. We can chat tomorrow.